cli/ingress-controller/main.go

```diff
@@ -241,9 +242,25 @@ func main() {
 		log.Fatalf("failed to create kong client: %v", err)
 	}
 
-	root, err := kongClient.Root(context.Background())
-	if err != nil {
-		log.Fatalf("failed to fetch metadata from kong: %v", err)
+	var root map[string]interface{}
+	backoff := flowcontrol.NewBackOff(1*time.Second, 15*time.Second)
+	const backoffID = "kong-admin-api"
+	retryCount := 0
+	for {
+		root, err = kongClient.Root(context.Background())
+		if err != nil {
+			if retryCount > 5 {
+				klog.Fatalf("failed to fetch metadata from kong: %v", err)
+			} else {
+				backoff.Next(backoffID, backoff.Clock.Now())
+				delay := backoff.Get(backoffID)
+				time.Sleep(delay)
+				retryCount++
+				klog.Infof("retry %d to fetch metadata from kong: %v", retryCount, err)
+				continue
+			}
+		}
+		break
```

Comment:
1. Please use log instead of klog. log is based on logrus, which provided structured logging for this project.
2. The branching could be made more readable:

```
for {
 root, err := kongClient...
  if err == nil {
    break
  }
  if retryCount > 5 {
    klog.Fatalf
  }
  // at this point, there is an error and a retry should be attempted
  
  // log the err
  // backoff logic
}
```
We should try to use minimum number of break and continue statements to avoid edge cases when a change is made to the code in future.